File: src/test/java/br/com/leonardoferreira/jirareport/factory/CreateBoardRequestFactory.java
package br.com.leonardoferreira.jirareport.factory;

import br.com.leonardoferreira.jbacon.JBacon;
import br.com.leonardoferreira.jirareport.domain.request.CreateBoardRequest;
import com.github.javafaker.Faker;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class CreateBoardRequestFactory extends JBacon<CreateBoardRequest>  {

    @Autowired
    private Faker faker;

    @Override
    protected CreateBoardRequest getDefault() {
        CreateBoardRequest createBoardRequest = new CreateBoardRequest();

        createBoardRequest.setName(faker.lorem().word());
        createBoardRequest.setExternalId(faker.number().randomNumber());

        return createBoardRequest;
    }

    @Override
    protected CreateBoardRequest getEmpty() {
        return new CreateBoardRequest();
    }

    @Override
    protected void persist(final CreateBoardRequest createBoardRequest) {
        throw new UnsupportedOperationException();
    }
}

File: src/test/kotlin/br/com/jiratorio/factory/domain/request/CreateBoardRequestFactory.kt
package br.com.jiratorio.factory.domain.request

import br.com.jiratorio.domain.request.CreateBoardRequest
import br.com.leonardoferreira.jbacon.JBacon
import com.github.javafaker.Faker
import org.springframework.stereotype.Component

@Component
class CreateBoardRequestFactory(
    private val faker: Faker
) : JBacon<CreateBoardRequest>() {

    override fun getDefault() =
        CreateBoardRequest(
            name = faker.lorem().word(),
            externalId = faker.number().randomNumber()
        )

    override fun getEmpty() =
        CreateBoardRequest(
            name = faker.lorem().word(),
            externalId = faker.number().randomNumber()
        )

    override fun persist(createBoardRequest: CreateBoardRequest) =
        throw UnsupportedOperationException()

}
